jep/0000/README.adoc

```diff
@@ -0,0 +1,366 @@
+* Jenkins plugin development requires Java 11 or newer
+** Jenkins plugins are not required to support Java 8 if they require a Jenkins version that does not support Java 8
+* Building all components with Java 11
+* Jenkins replace older API calls with Java 11 API calls
+** Jakarta EE upgrade is not required as part of this JEP
+* All plugins are operational with Java 11
+** Some Jenkins components may be updated to support features offered in Java 11 and beyond, but there is no plan to update all tools
+* Multi-release jar support in development tools
+* Cleanup of removed or deprecated features
+
+=== Scope of changes
+
+* Jenkins core
+* Jenkins remoting
+* Jenkins configuration as code
+* Docker packaging
+* Plugins
+* Jenkins website: www.jenkins.io
+* Jenkins CI instances:
+    link:https://ci.jenkins.io/[ci.jenkins.io],
+    infra.ci.jenkins.io,
+    trusted.ci.jenkins.io,
+    link:https://www.jenkins.io/security/#team[Security team]'s instance
+* Maven build flow (
+    link:https://github.com/jenkinsci/maven-hpi-plugin[Maven HPI Plugin],
+    link:https://github.com/jenkinsci/plugin-pom[Plugin POM],
+    link:https://github.com/jenkinsci/pom[Jenkins POM],
+    link:https://github.com/jenkinsci/bom[Jenkins plugin bill of materials POM],
+    etc.)
+* Test tools:
+    link:https://github.com/jenkinsci/jenkins-test-harness[Jenkins Test Harness (JTH)],
+    link:https://github.com/jenkinsci/acceptance-test-harness[Acceptance Test Harness (ATH)],
+    link:https://github.com/jenkinsci/plugin-compat-tester[Plugin Compatibility Tester (PCT)]
+* User and Developer Documentation
+
+Out of the scope for this JEP:
+
+* Packaging in subprojects (unless specifically noted): Jenkins Operator, Jenkins Evergreen, Jenkinsfile Runner, etc.
+  They will be handled in follow-up JEPs if needed.
+* Gradle build flow
+
+=== Jenkins core patches
+
+Work to be considered is defined in link:https://issues.jenkins.io/browse/JENKINS-67688[JENKINS-67688].
+
+==== Library updates
+
+* The link:https://issues.jenkins.io/browse/JENKINS-67688[JENKINS-67688 epic] will include library updates as needed.
+* Some updates may require downstream plugin updates.
+
+==== Core patches
+
+* link:https://github.com/jenkinsci/jenkins/pull/6083[PR 6083] - Drop Java 8 from Jenkins core
+* link:https://github.com/jenkinsci/jenkins/pull/6086[PR 6086] - Switch Jenkins acceptance test harness in core to Java 11
+* link:https://github.com/jenkinsci/jenkins/pull/6092[PR 6092] - Add administrative monitor to remind of upcoming end of Java 8 support
+
+==== Build flow updates
+
+* Jenkinsfile is updated to stop running tests with Java 8
+** It includes Unit tests, JTH and ATH smoke tests
+* It is possible to build Jenkins Core with the release profile on Java 8
+
+=== Jenkins Docker packaging
+
+The containers tagged for Java 8, like `latest-jdk8` and `centos7-jdk8` will no longer be updated.
+The upgrade guide and the announcement blogpost will note that users must switch to other images.
```

Comment:
As a final label, the docker image could include a warning to tell users to switch to another label.

Reply:
Good suggestion.  I'm not sure how that would best be implemented.  

Would the Docker image add a new "flag file" to Jenkins home directory (this image is no longer being updated) and then the presence of that flag file would cause an administrative monitor to be displayed?

Is there a better way to get the message to the Jenkins administrator?

Reply:
I just thought about a plain warning message in stderr, but indeed, that could go further and be implemented as an administrative monitor as you suggest.

Reply:
there's already an admin monitor to tell them to switch and a pending PR to add an EOL date

Reply:
> I just thought about a plain warning message in stderr, but indeed, that could go further and be implemented as an administrative monitor as you suggest.

I fear that warnings to the stderr of a Jenkins controller will not be seen by most administrators.

Reply:
> there's already an admin monitor to tell them to switch and a pending PR to add an EOL date

I was thinking that this admin monitor would be informing the user that the Docker image is no longer being updated.  We could have used that for the end of life of the centos8 image and would need something similar for the end of life for the Debian 10 (buster) image when Debian 12 is released.

I guess that an admin monitor may not be well suited for that idea, since we would want a distinct admin monitor for each image end of life.  If we tried to use the same admin monitor for all image end of life messages, then the administrator may have dismissed the end of life for an earlier image and we need to show them that the current image they are using has reached the end of its life

Reply:
OK. Just not sure how to make it clear to people consuming images that they are out of support.

Reply:
> OK. Just not sure how to make it clear to people consuming images that they are out of support

I think you're right that we have a problem there.  We don't have a way to tell users that the Docker image they are using is no longer being updated.  Maybe we should consider a new administrative monitor each time we stop updates on a Docker container.  If Jenkins is running in a container that is no longer being updated, we tell the administrator at least once that they need to move to a different Docker label.